### ISSUE TITLE
Support for Rider

### ISSUE THREAD

**Issue by maxjoehnk** [testcop/Resharper.TestCop#38]
Support for the Rider IDE would be neat.
As far as I know exchanging the Resharper.Sdk with the Rider.Sdk should work for this.

**Comment by testcop** [comment]
Hi Max, I doubt it would be as simple as that because Testcop uses some VisualStudio routines and also has a configuration dialog.

**Comment by jcsonder** [comment]
It would be really nice to have TestCop for Rider!

**Comment by Hadda90** [comment]
I would also welcome this proposal.

**Comment by testcop** [comment]
pull requests accepted :)

**Comment by mu88** [comment]
Today I spent an hour to see how far I can push this plugin towards Rider. It is definitely not easy because there are some Visual Studio / WPF components that have to be replaced.

**Comment by testcop** [comment]
@mu88   you are star for trying to get it working !

**Comment by mu88** [comment]
Maybe a first good step would be to decouple the UI from Visual Studio 🤔 introducing some necessary interfaces would at least make it clear to a future hero what to implement within Rider.

**Comment by jcsonder** [comment]
Any progress on that? 😀

**Comment by mu88** [comment]
Yes and no 😉 Instead of pushing TestCop forward, I started supporting @vladyslav-burylov and his project [Test Linker 2](https://github.com/vladyslav-burylov/resharper-testlinker2). That one is compatible with both Rider and R# and does all I need. You can obtain it from the [JetBrains store](https://plugins.jetbrains.com/plugin/15518-test-linker-2).

**Comment by testcop** [comment]
Any reason for testcop when Test Linker 2 does the same?
BTW was it pretty straight forward to do the UI element (config settings) in Test Linker

**Comment by mu88** [comment]
Honestly I didn't use TestCop for quite some time, therefore I don't know whether TestCop and Test Linker 2 have feature parity. But for me it offers all I need.

Regarding the UI, the plugin originally started at a different point: @matkoch first implemented his version which was already based on the original JetBrains plugin template supporting both Rider and R#. Then he archived his project and @vladyslav-burylov took over and maintained the plugin.  
But as far as I can see, the plugin does not explicitly work with UI elements. It uses abstractions provided from the SDK. [Here](https://github.com/vladyslav-burylov/resharper-testlinker2/blob/master/src/dotnet/ReSharperPlugin.TestLinker2/Options/TestLinker2OptionsPage.cs) you can see the code for the options page.

**Comment by jcsonder** [comment]
Thanks for the hint, @mu88 .
But _Test Linker 2_ doesn't like big solutions. 😢 
I just see this 
![grafik](https://user-images.githubusercontent.com/7531115/155089302-973d8a22-d4f0-4f01-802d-6a842e3627c6.png)
and have to wait ages until I have
![grafik](https://user-images.githubusercontent.com/7531115/155090841-0a52b118-d5b3-48d9-b1d3-2a8a01f423fa.png)
Only with small solutions it works fine.

I have to find out if the _Test Linker 2_ guys already are working on that problem... 
or just hope that there will be a faster & better solution soon in TestCop. 😀

**Comment by mu88** [comment]
@jcsonder Feel free to file an issue (preferably with an example of a big Solution)
